src/components/TicketModal/TicketModal.vue

```diff
@@ -0,0 +1,272 @@
+          placeholder="mm / dd / yyyy"
+        />
+      </div>
+    </div>
+    <div class="ticket-modal-third-row columns">
+      <div class="ticket-description column">
+        <label class="ticket-field-label">Description</label>
+        <textarea
+          class="textarea has-fixed-size is-fullwidth"
+          placeholder="Description"
+        ></textarea>
+      </div>
+    </div>
+    <div class="ticket-modal-footer">
+      <button class="button is-success">Save changes</button>
+      <button class="button">Cancel</button>
+      <button class="button is-danger">Delete</button>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent, ref, computed } from "vue";
+import { ReadTicketRecord } from "@/api/types";
+import IconSluggo from "@/assets/IconSluggo";
+
+const ticketModalComponent = defineComponent({
+  name: "TicketModal",
+  components: {
+    IconSluggo
+  },
+  props: {
+    ticketRecord: {
+      type: Object as () => ReadTicketRecord
+    }
+  },
+  emits: ["close"],
+  created() {
+    window.addEventListener("keydown", e => {
+      if (e.key == "Enter") {
+        this.saveChanges();
+      }
+    });
+  },
+  setup: () => {
+    const shouldShowPencil = ref(true);
+    const ticketTitle = ref("Temp Title");
+    const ticketStatus = ref("In Progress");
+    const statusDropdownClass = ref("");
+    const userDropdownClass = ref("");
+    const teamDropdownClass = ref("");
+    const tagsDropdownClass = ref("");
+    const statusElement = ref<HTMLElement | null>(null);
+    const isEditing = ref(false);
+    const tempText = ref("");
+    const toggleStatusDropdown = () => {
+      statusDropdownClass.value =
+        statusDropdownClass.value == "is-active" ? "" : "is-active";
+    };
+    const toggleUserDropdown = () => {
+      userDropdownClass.value =
+        userDropdownClass.value == "is-active" ? "" : "is-active";
+    };
+    const toggleTeamDropdown = () => {
+      teamDropdownClass.value =
+        teamDropdownClass.value == "is-active" ? "" : "is-active";
+    };
+    const toggleTagsDropdown = () => {
+      tagsDropdownClass.value =
+        tagsDropdownClass.value == "is-active" ? "" : "is-active";
+    };
+    const setTicketStatus = (status: string) => {
+      statusDropdownClass.value = "";
+      ticketStatus.value = status;
+    };
+    const enableEditing = () => {
+      isEditing.value = true;
+      shouldShowPencil.value = false;
+      tempText.value = ticketTitle.value;
+      isEditing.value = true;
+    };
+    const disableEditing = () => {
+      isEditing.value = false;
+      shouldShowPencil.value = true;
+      tempText.value = "";
+      isEditing.value = false;
+    };
+    const saveChanges = () => {
+      if (isEditing.value == true) {
+        isEditing.value = false;
+        ticketTitle.value = tempText.value;
+      }
+    };
+    const computedStatusColor = computed(() => {
+      if (ticketStatus.value == "In Progress") {
```

Comment:
The statuses returned by the API should include the color that they should be rendered with. If the SPA code does not reflect statuses with colors, please update it.